test/hotspot/jtreg/gc/g1/TestSoftMaxHeapSize.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2025, Google LLC. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package gc.g1;
+
+/*
+ * @test
+ * @bug 8236073
+ * @requires vm.gc.G1 & vm.opt.ExplicitGCInvokesConcurrent != true
+ * @library /test/lib
+ * @run main/othervm -Xmx200m -XX:MinHeapSize=4m -XX:MinHeapFreeRatio=99
+        -XX:MaxHeapFreeRatio=99 gc.g1.TestSoftMaxHeapSize
+ * @summary SoftMaxHeapSize should limit G1's heap size when resizing.
+ */
+
+import java.lang.management.ManagementFactory;
+import java.util.LinkedList;
+import java.util.List;
+import jdk.test.lib.dcmd.PidJcmdExecutor;
+
+public class TestSoftMaxHeapSize {
+
+  private static final int OBJECT_SIZE = 1000;
+  private static final long ALLOCATED_BYTES = 20_000_000; // About 20M
+  private static final long MAX_HEAP_SIZE =
+      200 * 1024 * 1024; // 200MiB, must match -Xmx on command line.
```

Comment:
Is it possible to get that value from the `MemoryMXBean` instead of relying on manual update? I.e. `getMax()`?

Reply:
Yes, it is a good idea. Done.